mlir/lib/Dialect/Rock/Transforms/AlignTiling.cpp

```diff
@@ -741,6 +745,34 @@ static void reconfigureLAGeneric(LinalgAlignRewriter &b,
   laGeneric.setIteratorTypesAttr(ArrayAttr::get(ctx, iteratorTypes));
 }
 
+static LogicalResult canFuseAcrossAtomic(LinalgAlignRewriter &b,
+                                         linalg::GenericOp laGeneric) {
+  bool isLegal = true;
+  for (auto &region : laGeneric->getRegions()) {
+    for (auto &block : region) {
+      for (auto &op : block) {
+        bool isTruncOp = llvm::isa<arith::TruncFOp, arith::TruncIOp>(op);
+        if (isTruncOp) {
+          OpResult result = op.getResult(0);
+          bool allowedType = result.getType() == b.getF32Type();
+          allowedType |= result.getType() == b.getF16Type();
+          if (!allowedType) {
+            isLegal = false;
+            break;
+          }
+        }
+
+        const bool isYieldOp = llvm::isa<linalg::YieldOp>(op);
+        if (!(isTruncOp || isYieldOp)) {
+          isLegal = false;
+          break;
+        }
+      }
+    }
+  }
+  return isLegal ? mlir::success() : mlir::failure();
```

Comment:
1. Don't need the `mlir::` here.
2. Neat little trick `return success(isLegal);` saves you the if operatior.